backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceEventsHandler.java

```diff
@@ -0,0 +1,202 @@
+    {
+        // This should always have a nonnull looper because BacktraceHandlerThread starts in the
+        // constructor and getLooper blocks until the looper is ready if the thread is started
+        super(backtraceHandlerThread.getLooper());
+        if (backtraceHandlerThread.getLooper() == null) {
+            throw new NullPointerException("Expected nonnull looper, this should not happen");
+        }
+        this.context = context;
+        this.customAttributes = customAttributes;
+        this.backtraceHandlerThread = backtraceHandlerThread;
+        this.api = api;
+        this.submissionUrl = settings.getBaseUrl() + "/" + urlPrefix + "/submit?token=" + settings.getToken() + "&universe=" + settings.getUniverseName();
+        this.timeBetweenRetriesMillis = settings.getTimeBetweenRetriesMillis();
+
+        long timeIntervalMillis = settings.getTimeIntervalMillis();
+
+        if (timeIntervalMillis != 0) {
+            final BacktraceEventsHandler handler = this;
+            handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    handler.send();
+                    handler.postDelayed(this, timeIntervalMillis);
+                }
+            }, timeIntervalMillis);
+        }
+    }
+
+    public abstract void sendStartupEvent(String eventName);
+
+    public void send() {
+        sendEvents(events, 0);
+    }
+
+    private void sendEvents(ConcurrentLinkedDeque<Event> events, int attempts) {
+        if (events == null) {
+            return;
+        }
+        if (events.size() == 0) {
+            return;
+        }
+        try {
+            EventsPayload payload = getEventsPayload();
+            api.sendEventsPayload(payload);
+        } catch (Exception e) {
+            BacktraceLogger.e(LOG_TAG, "Could not create events payload for metrics submission");
+            return;
+        }
+    }
+
+    @Override
+    public void handleMessage(Message msg) {
+        final BacktraceHandlerInputEvents input = (BacktraceHandlerInputEvents) msg.obj;
+        EventsResult result;
+
+        if (input.eventsRequestHandler != null) {
+            BacktraceLogger.d(LOG_TAG, "Sending using custom request handler");
+            result = input.eventsRequestHandler.onRequest(input.payload);
+        } else {
+            BacktraceLogger.d(LOG_TAG, "Sending report using default request handler");
+            String json = BacktraceSerializeHelper.toJson(input.payload);
+            result = BacktraceReportSender.sendEvents(submissionUrl, json, input.payload, input.serverErrorEventListener);
+        }
+
+        if (input.eventsOnServerResponseEventListener != null) {
+            BacktraceLogger.d(LOG_TAG, "Processing result using custom event");
+            input.eventsOnServerResponseEventListener.onEvent(result);
+        }
+
+        int statusCode = result.getStatusCode();
+        if (statusCode == 200) {
```

Comment:
you can use const `HttpURLConnection.HTTP_OK` and others here and below

Reply:
Sounds good, I will use constants for HTTP codes where available!